android-components/components/support/remotesettings/src/test/java/mozilla/components/support/remotesettings/RemoteSettingsClientTest.kt

```diff
@@ -0,0 +1,121 @@
+package mozilla.components.support.remotesettings
+
+import kotlinx.coroutines.runBlocking
+import mozilla.appservices.remotesettings.RemoteSettings
+import mozilla.appservices.remotesettings.RemoteSettingsRecord
+import org.json.JSONObject
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.`when`
+import org.robolectric.RobolectricTestRunner
+import java.io.File
+import java.io.FileReader
+import java.io.IOException
+
+@RunWith(RobolectricTestRunner::class)
+class RemoteSettingsClientTest {
+
+    private lateinit var mockRemoteSettings: RemoteSettings
+    private lateinit var mockFile: File
+    private lateinit var mockFileReader: FileReader
+    private lateinit var mockJsonObject: JSONObject
+    private lateinit var remoteSettingsClient: RemoteSettingsClient
+
+    @Before
+    fun setUp() {
+        mockFile = mock(File::class.java)
+        mockFileReader = mock(FileReader::class.java)
+        mockJsonObject = mock(JSONObject::class.java)
+        mockRemoteSettings = mock(RemoteSettings::class.java)
+        remoteSettingsClient = RemoteSettingsClient(File(""), "https://firefox.settings.services.mozilla.com", "", "")
+    }
+
+    @Test
+    fun `GIVEN empty records WHEN write is called THEN the result is disk failure`() {
+        val records = emptyList<RemoteSettingsRecord>()
+
+        val result = runBlocking { remoteSettingsClient.write(records) }
+
+        assertTrue(result is RemoteSettingsResult.DiskFailure)
+    }
+
+    fun `GIVEN non-empty records WHEN write is called THEN the result is a success`() {
+        val tempFile = File.createTempFile("test", ".json")
+        val records = listOf(
+            RemoteSettingsRecord("1", 123u, false, null, JSONObject()),
+            RemoteSettingsRecord("2", 456u, true, null, JSONObject()),
+        )
+
+        val result = runBlocking { RemoteSettingsClient(tempFile).write(records) }
+
+        require(result is RemoteSettingsResult.Success) { "Result should be Success" }
+        assertEquals(records, result.records)
+
+        tempFile.delete()
+    }
+
+    @Test
+    fun `test fetchAndWrite success`() {
+        val client = mock(RemoteSettingsClient::class.java)
```

Comment:
nit: For inline mocks like these, I think the syntax that uses generics is more idiomatic
```suggestion
        val client = mock<RemoteSettingsClient>()
```